comit/src/float_math.rs

```diff
@@ -0,0 +1,223 @@
+use bitcoin::hashes::core::cmp::Ordering;
+use num::{BigUint, Zero};
+use std::str::FromStr;
+
+/// Truncate the float's mantissa to length `precision`.
+pub fn truncate(float: f64, precision: u16) -> f64 {
+    let s = float.to_string();
+    truncate_str(&s, precision)
+}
+
+/// Truncate the float's mantissa to length `precision`.
```

Comment:
```suggestion
/// Truncate the number's mantissa to length `precision`.
```

I have loosely used the word `float` when I actually mean `number` or more precisely `fractional number`.

Reply:
How about we favour the term `fraction` in identifiers if we choose to use that as the filename?